### PR TITLE
C++11 cleanup, p2: Remove ScaledMatrix

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #4212

### CODE REVIEW

**Comment by bangerth** [comment]
Rather than trying to throw everything into one big patch, why not submit multiple PRs that are all easier to review?

**Comment by tamiko** [comment]
@bangerth Even though it might look like much, this changeset is pretty self-contained.

**Comment by bangerth** [comment]
Yes, but it's still a lot of code to look over :-)

I think I thought from your original PR message that you wanted to remove a whole bunch of other classes as well as part of this PR. But it looks like all that's missing is just checking the testsuite and writing a changelog blurb?

**Comment by tamiko** [comment]
@bangerth Just these classes. I can split anyway.

**Comment by tamiko** [comment]
@bangerth Voila.

**Comment by bangerth** [comment]
Ah, I didn't mean to cause you that much more work. I just think that in general smaller patches are easier to review (and revert). Anyway, thanks for breaking things up!

**Comment by tamiko** [comment]
@bangerth It's fine. It is a bit more methodical and did cost me only 10-12 minutes more work.

File: doc/news/changes/incompatibilities/20170406MatthiasMaier [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Removed: A number of deprecated classes have been removed in favor of the
+new LinearOperator concept: TODO
```
Comment (tamiko) [inline doc/news/changes/incompatibilities/20170406MatthiasMaier]: @bangerth This is the placeholder for the detailed description of every retired class and its replacement.

Comment (bangerth) [inline doc/news/changes/incompatibilities/20170406MatthiasMaier]: OK.

**Comment by bangerth** [comment]
There's a merge conflict. Can you update?

**Comment by tamiko** [comment]
@bangerth Rebased.

**Comment by bangerth** [comment]
OK if it passes the tester.

**Pull request merged**